File: src/main/java/com/e16din/lightutils/IsIt.java
package com.e16din.lightutils;

import android.support.annotation.NonNull;
import android.util.Patterns;

import java.util.regex.Pattern;

/**
 * Created by e16din on 02.09.15.
 */
public final class IsIt {
    private IsIt() {
    }

    public static boolean phone(@NonNull CharSequence s, int countryId) {
        switch (countryId) {
            case Countries.ID_RUSSIA:
                //todo
                break;
            case Countries.ID_NONE:
            default:
                Pattern pattern = Patterns.PHONE;
                return pattern.matcher(s).matches();
        }

        return false;
    }

    public static boolean phone(@NonNull CharSequence s) {
        return phone(s, Countries.ID_NONE);
    }

    public static boolean email(@NonNull CharSequence s) {
        Pattern pattern = Patterns.EMAIL_ADDRESS;
        return pattern.matcher(s).matches();
    }

    public static boolean number(@NonNull String s) {
        try {
            Integer.parseInt(s);
            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }

    public static boolean floatNumber(@NonNull String s) {
        try {
            Double.parseDouble(s);
            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }

    public static boolean nil(Object obj) {
        return obj == null;
    }
}

File: src/main/java/com/e16din/lightutils/IsIt.kt
package com.e16din.lightutils

import android.telephony.PhoneNumberUtils
import android.util.Patterns
import com.e16din.lightutils.model.Countries

/**
 * Created by e16din on 02.09.15.
 */
object IsIt {

    @JvmOverloads
    fun phone(phoneStr: String, countryId: Int = Countries.ID_NONE) =
            (phoneStr.length in 7..12 && PhoneNumberUtils.isGlobalPhoneNumber(phoneStr))

    @JvmOverloads
    fun email(s: CharSequence) = Patterns.EMAIL_ADDRESS.matcher(s).matches()

    @JvmOverloads
    fun number(s: String) = (s.toDoubleOrNull() != null)

    @JvmOverloads
    fun nil(obj: Any?) = (obj == null)
}
